UserStudentApp/views.py

```diff
@@ -0,0 +1,87 @@
+from .models import Student, CustomUser, Course, UserImage
+from .serializers import StudentSerializer, UserSerializer, CourseSerializer, ImageSerializer
+from rest_framework import permissions,viewsets,status
+from .permissions import IsOwnerOrReadOnly
+from rest_framework.response import Response
+from django.db.models import Q
+
+
+class UserViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    This viewset automatically provides `list` and `retrieve` actions.
+    """
+    permission_classes  = [IsOwnerOrReadOnly,permissions.IsAuthenticated]
+    serializer_class    = UserSerializer
+
+
+    def get_queryset(self):
+        request_user = self.request.user
+        name         = self.request.query_params.get('name')
+        
+        if request_user.is_superuser:
+            queryset = CustomUser.objects.all()
+            if name:
+                queryset = CustomUser.objects.filter(username= name) 
+        else:
+            queryset = CustomUser.objects.filter(email= request_user)
+
+        return queryset
+  
+
+class StudentViewSet(viewsets.ModelViewSet):
+    """
+    This viewset automatically provides `list`, `create`, `retrieve`,
+    `update` and `destroy` actions.
+    """
+
+    permission_classes = [IsOwnerOrReadOnly,permissions.IsAuthenticated]
+    serializer_class   = StudentSerializer
+
+
+    def get_queryset(self, *args, **kwargs):
+        request_user = self.request.user
+        if request_user.is_superuser:
+            queryset = Student.objects.all()
+            names    = self.request.query_params.getlist('name')
+            if names:
```

Comment:
if `names` is not empty list why are you first getting all() and then applying filter on the basis of name?

Reply:
Since we are superuser, so a superuser can we view all the data so thats why first he has all the students then if there is a name or list of names then we can override queryset that contains all students to the names the students in list.

Reply:
thats what i want to change. why are you fetching all the records if there is a name? why are hitting 2 queries without any reason.

Reply:
I have changed it for both User and StudentViewSet. So now we will hit DB only one time.